runtime/port/zos390/j9mprotect.s

```diff
@@ -47,5 +47,4 @@ _MPROT   CELQPRLG BASEREG=8
          IHAPVT
          LTORG
  
-         END
-
+         END
```

Comment:
Is this change that removes the last newline in the file necessary?
Same with j9numprotect.s

Reply:
Yes, it seems to trigger this error owing to that new line. Only when removing it, does that go away.
```
ASMA140W END record missing
 ASMA435I Record 51 in /jit/team/gauravc/repos/openj9-openjdk-jdk21-zos/openj9/runtime/port/zos390/j9mprotect.s on
       volume:
FSUM3401 The assemble step ended with rc = 4.
ibm-clang: error: assembler command failed with exit code 4 (use -v to see invocation)
```

Reply:
I see that j9gs_store_gscb.s and j9gs_load_gscb.s have a comma after `END` before the final newline. Is that an alternative to removing the newline?

Reply:
Yes, it seems to trigger this error owing to that new line. Only when removing it, does that go away.
```
ASMA140W END record missing
 ASMA435I Record 51 in /openj9/runtime/port/zos390/j9mprotect.s on
       volume:
FSUM3401 The assemble step ended with rc = 4.
ibm-clang: error: assembler command failed with exit code 4 (use -v to see invocation)
```

Reply:
From https://www.ibm.com/docs/en/zos/2.2.0?topic=SSLTBW_2.2.0/com.ibm.zos.v2r2.asma400/end.html, it notes:

> The END instruction must always be the last statement in the source program. 

and 

> If the END instruction is omitted, one is generated by the assembler, and message ASMA140W END record missing is issued.

Interestingly, we did get that `ASMA140W END record missing` above.  I'm a bit surprised by a new line will cause issues, but it is complaining about line 51, which is that that last line after END.

Reply:
Processing by `m4` seems to add an extra newline at the end and the compiler/assembler complain if the last line doesn't contain `END`. Another idea would be to tweak `j9vm_gen_asm` so it removes a trailing (or all) blank line(s).
```
			COMMAND m4 ${m4_includes} ${m4_defines} ${CMAKE_CURRENT_SOURCE_DIR}/${m4_file} | grep -v '^$' > ${base_name}.s
```

Reply:
I much prefer a (perhaps z/OS-specific) adjustment like that: our line-end check is likely to trigger someone to put the final newline back in files like that (breaking the build).

We should also ask the compiler team why `END`, followed only by blank lines, is worthy of that warning/error.

Reply:
- Removed changes to `j9mprotect.s`  and `j9munprotect.s` (resolved via #19720)
- Added commit https://github.com/eclipse-openj9/openj9/pull/19524/commits/83ef23014a7fcc624aed98cc75a3f0afc8ce2501 to address the generated assembly file issue @keithc-ca